### PR TITLE
Migration - add pil licence number to profile

### PR DESCRIPTION
**Pull request by joefitter**
* added migration to add pil_licence_number field to profile
* added migration to populate value from most recently issued pil

### CODE REVIEW

File: migrations/20200824182434_migrate_pil_licence_no_to_profile.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    // get all profilesIds with a pil
+    .then(() => knex('pils').distinct('profile_id'))
+    .then(pilProfiles => {
+      return pilProfiles.reduce((promise, pilProfile) => {
+        return promise
+          .then(() => knex('profiles')
+            .select('id')
+            .where({ id: pilProfile.profile_id })
+            .whereNull('pil_licence_number')
+            .first()
```
Comment (lennym) [inline migrations/20200824182434_migrate_pil_licence_no_to_profile.js:12]: What is this query for? We use the profile id as the input, and then the only bit of the result we use is the profile id. It seems a little redundant.

Comment (lennym) [reply migrations/20200824182434_migrate_pil_licence_no_to_profile.js:12]: If the answer is "it checks that the profile doesn't _already_ have a PIL id" then you'll want to set all of them to null in the `rollback`.

File: schema/training-pil.js [hunk 2]
```diff
@@ -25,14 +25,28 @@ class TrainingPil extends BaseModel {
...
+      .withGraphFetched('profile');
+
+    if (sort.column) {
+      query = this.orderBy({ query, sort });
+    }
+
+    query = this.paginate({ query, limit, offset });
+
+    return query;
+  }
```
Comment (lennym) [inline schema/training-pil.js:40]: Is this meant to be here?

Comment (joefitter) [reply schema/training-pil.js:40]: probably not, I forgot I wasn't on master. I can pull this out

File: migrations/20200824182434_migrate_pil_licence_no_to_profile.js (hunk 1 above)
Comment (joefitter) [reply migrations/20200824182434_migrate_pil_licence_no_to_profile.js:12]: you are right on both counts, I was actually going to do that but got distracted by something shiny

**Pull request merged**